processing/src/main/java/org/apache/carbondata/processing/loading/sort/unsafe/holder/UnsafeSortTempFileChunkHolder.java

```diff
@@ -312,7 +217,7 @@ private void fillDataForPrefetch() {
       }
 
       // write complex dimensions here.
-      for (; dimCount < dimensionCount; dimCount++) {
+      for (; dimCount < sortParameters.getComplexDimColCount(); dimCount++) {
```

Comment:
I think there is difference with changed code compare to old code. Please double check the changes. Complex dimensions may not work

Reply:
yeah, I made a mistake when applying the changes to this PR.